File: 1-two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = HashMap<Int, Int>()

        for (i in nums.indices) {
            val num = nums[i]
            if (map.containsKey(num)) {
                return intArrayOf(map.getOrDefault(num, 0), i)
            }
            else {
                map.put(target - num, i)    
            }
        }

        return intArrayOf(0, 0)
    }
}

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {}

        for i,num in enumerate(nums):
            if num in map:
                return [map[num], i]
            else:
                map[target - num] = i     
        